model.py

```diff
@@ -180,19 +180,23 @@ def __init__(self,in_channels,h_1,h_2,h_3,h_4,drop_out,device):
     self.lin4   = nn.Linear(h_3,h_4,bias=True)
     self.lin5   = nn.Linear(h_4,1,bias=True)
 
-    self.act        = nn.Tanh()
-    self.act_last   = nn.Sigmoid()
+    self.act1        = nn.Tanh()
+    self.act2        = nn.Tanh()
+    self.act3        = nn.Tanh()
+    self.act4        = nn.Tanh()
+    self.act5        = nn.Tanh()

```

Comment:
You could use the same activation object instead of creating many. This is similar to here:
https://github.com/pytorch/vision/blob/master/torchvision/models/resnet.py#L72-L81

Reply:
I made them separate, so I can track activations from different layers. I will have a look again how to use PyTorch hooks